lms/models/upload.py

```diff
@@ -51,10 +55,17 @@ def _run_auto_checks(solution: Solution) -> None:
 
 
 def new(user: User, file: FileStorage) -> Tuple[List[int], List[int]]:
+    solution_hash = hashing.by_file(file)
+
     matches: List[int] = []
     misses: List[int] = []
     errors: List[Union[UploadError, AlreadyExists]] = []
+
     for exercise_id, files, solution_hash in Extractor(file):
+        exercise: Exercise = Exercise.get_or_none(exercise_id)
+        if _is_uploaded_before(user, exercise, solution_hash):
+            raise AlreadyExists('You try to reupload an old solution.')
```

Comment:
Look at the `upload_page` function in `views.py`.
You raise here an error that we don't handle there :)